packages/flutter_tools/lib/src/project.dart

```diff
@@ -509,30 +509,33 @@ class AndroidProject extends FlutterProjectPlatform {
     if (plugin.existsSync()) {
       return false;
     }
-    final File appGradle = hostAppGradleRoot.childFile(
-        fileSystem.path.join('app', 'build.gradle'));
-    if (!appGradle.existsSync()) {
-      return false;
-    }
-    for (final String line in appGradle.readAsLinesSync()) {
-      final bool fileBasedApply = line.contains(RegExp(r'apply from: .*/flutter.gradle'));
-      final bool declarativeApply = line.contains('dev.flutter.flutter-gradle-plugin');
-      final bool managed = line.contains("def flutterPluginVersion = 'managed'");
-      if (fileBasedApply || declarativeApply || managed) {
-        return true;
+    try {
+      for (final String line in appGradleFile.readAsLinesSync()) {
+        final bool fileBasedApply = line.contains(RegExp(r'apply from: .*/flutter.gradle'));
+        final bool declarativeApply = line.contains('dev.flutter.flutter-gradle-plugin');
+        final bool managed = line.contains("def flutterPluginVersion = 'managed'");
+        if (fileBasedApply || declarativeApply || managed) {
+          return true;
+        }
       }
+    } on FileSystemException {
```

Comment:
Should this log/report an error?

Reply:
This pattern is just following the previous behavior (see the old line 515) where if the file does not exist, we return false (i.e. no change other than removing race condition). 

I can add a log though.